drivers/counter/counter_ll_stm32_rtc.c

```diff
@@ -179,19 +210,37 @@ static int rtc_stm32_set_alarm(const struct device *dev, uint8_t chan_id,
 	rtc_alarm.AlarmMask = LL_RTC_ALMA_MASK_NONE;
 	rtc_alarm.AlarmDateWeekDaySel = LL_RTC_ALMA_DATEWEEKDAYSEL_DATE;
 	rtc_alarm.AlarmDateWeekDay = alarm_tm.tm_mday;
+#else
+	rtc_alarm.AlarmTime.Hours = remain / 3600;
+	remain -= rtc_alarm.AlarmTime.Hours * 3600;
+	rtc_alarm.AlarmTime.Minutes = remain / 60;
+	remain -= rtc_alarm.AlarmTime.Minutes * 60;
+	rtc_alarm.AlarmTime.Seconds = remain;
+#endif
 
+#if !defined(CONFIG_SOC_SERIES_STM32F1X)
 	LL_RTC_DisableWriteProtection(RTC);
 	LL_RTC_ALMA_Disable(RTC);
 	LL_RTC_EnableWriteProtection(RTC);
+#endif
 
+#if !defined(CONFIG_SOC_SERIES_STM32F1X)
 	if (LL_RTC_ALMA_Init(RTC, LL_RTC_FORMAT_BIN, &rtc_alarm) != SUCCESS) {
+#else
+	if (LL_RTC_ALARM_Init(RTC, LL_RTC_FORMAT_BIN, &rtc_alarm) != SUCCESS) {
+#endif
 		return -EIO;
 	}
 
 	LL_RTC_DisableWriteProtection(RTC);
+#if !defined(CONFIG_SOC_SERIES_STM32F1X)
 	LL_RTC_ALMA_Enable(RTC);
 	LL_RTC_ClearFlag_ALRA(RTC);
 	LL_RTC_EnableIT_ALRA(RTC);
+#else
+	LL_RTC_ClearFlag_ALR(RTC);
+	LL_RTC_EnableIT_ALR(RTC);
+#endif
```

Comment:
Would you mind adding a function to factorize this piece of code and minimize the number of `#if !defined(CONFIG_SOC_SERIES_STM32F1X) `.
This will help for code readbility

Reply:
I change it.